Explorer/Assets/DCL/SceneLoadingScreens/SceneLoadingScreenController.cs

```diff
@@ -0,0 +1,164 @@
+using Cysharp.Threading.Tasks;
+using DCL.Utilities.Extensions;
+using MVC;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using UnityEngine;
+using UnityEngine.Localization.SmartFormat.PersistentVariables;
+using UnityEngine.Pool;
+using Utility;
+
+namespace DCL.SceneLoadingScreens
+{
+    public partial class SceneLoadingScreenController : ControllerBase<SceneLoadingScreenView, SceneLoadingScreenController.Params>
+    {
+        private readonly ISceneTipsProvider sceneTipsProvider;
+
+        private int currentTip;
+        private SceneTips tips;
+        private CancellationTokenSource? tipsRotationCancellationToken;
+        private IntVariable? progressLabel;
+
+        public SceneLoadingScreenController(ViewFactoryMethod viewFactory,
+            ISceneTipsProvider sceneTipsProvider) : base(viewFactory)
+        {
+            this.sceneTipsProvider = sceneTipsProvider;
+        }
+
+        public override CanvasOrdering.SortingLayer Layer => CanvasOrdering.SortingLayer.Overlay;
+
+        public override void Dispose()
+        {
+            base.Dispose();
+
+            tipsRotationCancellationToken?.SafeCancelAndDispose();
+        }
+
+        protected override void OnViewInstantiated()
+        {
+            base.OnViewInstantiated();
+
+            viewInstance.ShowNextButton.onClick.AddListener(() =>
+            {
+                ShowTip(currentTip + 1);
+                tipsRotationCancellationToken = tipsRotationCancellationToken.SafeRestart();
+                RotateTipsOverTime(tips.Duration, tipsRotationCancellationToken.Token).Forget();
+            });
+
+            viewInstance.ShowPreviousButton.onClick.AddListener(() =>
+            {
+                ShowTip(currentTip - 1);
+                tipsRotationCancellationToken = tipsRotationCancellationToken.SafeRestart();
+                RotateTipsOverTime(tips.Duration, tipsRotationCancellationToken.Token).Forget();
+            });
+
+            progressLabel = (IntVariable)viewInstance.ProgressLabel.StringReference["progressValue"];
+        }
+
+        protected override void OnBeforeViewShow()
+        {
+            base.OnBeforeViewShow();
+
+            viewInstance.ProgressBar.normalizedValue = 0f;
+            progressLabel!.Value = 0;
+            viewInstance.ClearTips();
+        }
+
+        protected override void OnViewClose()
+        {
+            base.OnViewClose();
+
+            tipsRotationCancellationToken?.SafeCancelAndDispose();
+        }
+
+        protected override UniTask WaitForCloseIntent(CancellationToken ct)
+        {
+            return ShowTipsAsync(ct)
+                  .ContinueWith(() =>
```

Comment:
`ContinueWith` produces a delegate capture. Just make the whole method `async` and use sequential `await`: it's more efficient

Reply:
Converting this method into `async` forces me to rename it `WaitForCloseIntentAsync` otherwise code standards unit tests will fail. Is that ok?